common/src/com/thoughtworks/go/domain/materials/git/GitCommand.java

```diff
@@ -99,9 +101,18 @@ public Modification latestModification() {
             throw new RuntimeException(String.format("Working directory: %s\n%s", workingDir, outputStreamConsumer.getStdError()), e);
         }
 
-        CommandLine gitCmd = git(environment).withArg("log").withArg(remoteBranch()).withArgs(args).withWorkingDir(workingDir);
+        CommandLine gitCmd = git(environment).withArg("log").withArgs(args).withWorkingDir(workingDir);
+        ConsoleResult result;
+        try {
+            result = runOrBomb(gitCmd);
+        } catch (CommandLineException exception) {
+            if (exception.getMessage().contains("Invalid revision range")) {
```

Comment:
I'd not assume this message, it's prone to change in the future versions of git (and may be different in previous versions).

This message may also be different based on the locale and language settings that's being used and git may print this message in French or Russion for all you know :)